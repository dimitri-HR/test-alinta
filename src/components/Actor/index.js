import React, { Fragment } from 'react'

import { actorTypes } from '../../types'
import Roles from '../Roles'

const Actor = ({ actor: { name, roles } }) => (
  <Fragment>
    <p>{name}</p>
    <ul>
      <Roles roles={roles} />
    </ul>
  </Fragment>
)

Actor.propTypes = {
  actor: actorTypes.isRequired,
}

export default Actor
