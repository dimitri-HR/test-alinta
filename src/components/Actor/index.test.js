import React from 'react'
import renderer from 'react-test-renderer'

import Actor from '.'

const mockActor = {
  name: 'Actor',
  roles: [],
}

describe('Actor', () => {
  it('renders without errors', () => {
    const component = renderer.create(<Actor actor={mockActor}/>)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
