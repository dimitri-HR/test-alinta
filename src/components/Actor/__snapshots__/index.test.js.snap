// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actor renders without errors 1`] = `
Array [
  <p>
    Actor
</p>,
  <ul />,
]
`;
